libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -78,3 +142,546 @@ void test_MQTT_Init_Invalid_Params( void )
+    bool expectMoreCalls = true;
+
+    incomingPacket.type = currentPacketType;
+    incomingPacket.remainingLength = MQTT_SAMPLE_REMAINING_LENGTH;
+    MQTT_GetIncomingPacketTypeAndLength_ExpectAnyArgsAndReturn( modifyIncomingPacketStatus );
+    /* Replace pointer parameter being passed to the method. */
+    MQTT_GetIncomingPacketTypeAndLength_ReturnThruPtr_pIncomingPacket( &incomingPacket );
+
+    /* More calls are expected only with the following packet types. */
+    if( ( currentPacketType != MQTT_PACKET_TYPE_PUBLISH ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_PUBACK ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_PUBREC ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_PUBREL ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_PUBCOMP ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_PINGRESP ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_SUBACK ) &&
+        ( currentPacketType != MQTT_PACKET_TYPE_UNSUBACK ) )
+    {
+        expectMoreCalls = false;
+    }
+
+    /* When no data is available, the process loop tries to send a PINGREQ. */
+    if( modifyIncomingPacketStatus == MQTTNoDataAvailable )
+    {
+        if( ( pContext->waitingForPingResp == false ) &&
+            ( pContext->keepAliveIntervalSec != 0U ) )
+        {
+            MQTT_GetPingreqPacketSize_ExpectAnyArgsAndReturn( MQTTSuccess );
+            /* Replace pointer parameter being passed to the method. */
+            MQTT_GetPingreqPacketSize_ReturnThruPtr_pPacketSize( &pingreqSize );
+            MQTT_SerializePingreq_ExpectAnyArgsAndReturn( serializeStatus );
+        }
+
+        expectMoreCalls = false;
+    }
+
+    /* Deserialize based on the packet type (PUB or ACK) being received. */
+    if( expectMoreCalls )
+    {
+        if( incomingPublish )
+        {
+            MQTT_DeserializePublish_ExpectAnyArgsAndReturn( deserializeStatus );
+        }
+        else
+        {
+            MQTT_DeserializeAck_ExpectAnyArgsAndReturn( deserializeStatus );
+        }
+
+        if( ( deserializeStatus != MQTTSuccess ) ||
+            ( currentPacketType == MQTT_PACKET_TYPE_PINGRESP ) ||
+            ( currentPacketType == MQTT_PACKET_TYPE_SUBACK ) ||
+            ( currentPacketType == MQTT_PACKET_TYPE_UNSUBACK ) )
+        {
+            expectMoreCalls = false;
+        }
+    }
+
+    /* Update state based on the packet type (PUB or ACK) being received. */
+    if( expectMoreCalls )
+    {
+        if( incomingPublish )
+        {
+            MQTT_UpdateStatePublish_ExpectAnyArgsAndReturn( stateAfterDeserialize );
+        }
+        else
+        {
+            MQTT_UpdateStateAck_ExpectAnyArgsAndReturn( stateAfterDeserialize );
+        }
+
+        if( stateAfterDeserialize == MQTTPublishDone )
+        {
+            expectMoreCalls = false;
+        }
+    }
+
+    /* Serialize the packet to be sent in response to the received packet.
+     * Observe that there is no reason to serialize a PUBLISH after receiving
```

Comment:
I don't understand the significance of the second sentence in this comment block.

Reply:
Removing